packages/loader/container-loader/src/container.ts

```diff
@@ -514,9 +514,27 @@ export class Container
 		// It's conceivable the container could be closed when this is called
 		// Only transition states if currently loading
 		if (this._lifecycleState === "loading") {
-			// Propagate current connection state through the system.
-			this.propagateConnectionState(true /* initial transition */);
 			this._lifecycleState = "loaded";
+
+			// Propagate current connection state through the system.
+			const readonly = this.readOnlyInfo.readonly ?? false;
+			assert(this.connectionState !== ConnectionState.Connected, "not connected yet");
+			// This call does not look like needed any more, with delaying all connection-related events past loaded phase.
```

Comment:
nit: look like it is needed

Reply:
No, it's actually probably not needed, but I'm a bit afraid to remove it here, maybe in next PR.
It was always here (in a form of this.propagateConnectionState()).